core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -475,10 +475,14 @@ public void commit() {
     }
   }
 
+  Object updateEvent(Snapshot committedSnapshot) {
```

Comment:
`PendingUpdate.updateEvent` only usage is in SnapshotProducer currently so could change the interface directly to `updateEvent(Snapshot committedSnapshot)`, but did not want to make a backwards incompatible API change

The whole update event/listener functionality seems untouched for years.